adventure/adventure.py

```diff
@@ -3528,6 +3750,144 @@ def get_lock(self, member: discord.Member):
                 pass
             await self._update_hero(user, c)
 
+    async def _ability2_bonus(self, user, stat_checks, ability2_list_users):
+        bonus_stat = 0
+        try:
+            c = await Character._from_json(self.config, user)
+        except Exception:
+            log.error("Error with the new character sheet", exc_info=True)
+        continue
+        for stat in stat_checks: #calculates the standard stat value 
+            bonus_stat += getattr(c, stat) + c.skill[stat]
+        bonus_stat = int(bonus_stat / len(stat_checks))
+        if c.heroclass["ability2"]: #if ability2 is active, calculates the max stat)
+            max_stat = max(c.att + c.skill["att"], c.int + c.skill["int"], c.cha + c.skill["cha"], int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3))
+            if (bonus_stat - max_stat) == 0: #compare standard stat value with max stat
+                bonus_stat = bonus_stat * 1.1 #add a small bonus when the player uses ability2 with the atk type based on his max stat
+            else:
+                bonus_stat = max_stat
+            if stat_checks == ["att"]:
+                choice = "🗡"
+            elif stat_checks == ["int"]:
+                choice = "🌟"
+            elif stat_checks == ["cha"]:
+                choice = "🗨"
+            else:
+                choice = "🛐"
+            user_bonus = f"{bold(self.E(user.display_name))} ({choice}{bonus_stat})" 
+            ability2_list_users.append([heroclass["name"], user_bonus]) #adds the [class, user (bonus)] in a list for further display
+        return bonus_stat, ability2_list_users
+
+    async def _ability2_txt(self, type, ability2_list_users):
```

Comment:
"type" is a built-in method... can we rename it everywhere to like "attack_type"?